### PR TITLE
Add Velero mixin

### PR DESCRIPTION
**Pull request by ishanjainn**
The Velero mixin contains the following dashboards:

- Velero

and the following alerts:

- VeleroBackupPartialFailure
- VeleroBackupFailed

Screenshot 2023-07-12 at 4 12 41 PM

### CODE REVIEW

**Review by rgeyer — COMMENTED** [review]
More high-level design feedback here.

For all of the backup metrics, they are counters (except total backups, [which is a gauge](https://github.com/vmware-tanzu/velero/blame/main/pkg/metrics/metrics.go#L110), tho I suspect that's a bug in the exporter/app), meaning they only ever increase and should be storing the lifetime total value for each.

I would argue that these values aren't particularly useful as plain stats. So Velero has done 700 backups in it's lifetime. Over what period? On what schedule? Are there certain days of the week that backups are failing? Are those failures at odd times?

These values may be better shown on a timeseries, with all "like things" plotted on the same graph, with the `increase` function used over `$__rate_interval`.

Something like `increase(velero_backup_attempt_total{...}[$__rate_interval])`. This will show the number of backup attempts that occurred on a rolling window the size of `$__rate_interval` (roughly 5min with Grafana Cloud defaults, but this is calculated for you, see [this blog post](https://grafana.com/blog/2020/09/28/new-in-grafana-7.2-__rate_interval-for-prometheus-rate-queries-that-just-work/)), during the selected range.

For example, if you selected a time range of 2 days, you would see a timeseries with sawtooth "spikes" of the number of attempted backups during the 5min window they were attempted.

If you added all the other dimensions (deletions, and failures) on the same timeseries you could easily visualize all of the activity for a backup, snapshot, or CSI snapshot from it's corresponding timeseries.

If you wanted to keep the stats, I'd probably put them on a bar or pie graph, and increase with range to show the total increase over the selected timerange. I.E. It would show that there were 10 backups over the last 3 days selected by range. `increase(velero_backup_attempt_total{...}[$__range]`.

You may also want to add template variables to filter by [backup name and/or schedule](https://github.com/vmware-tanzu/velero/blame/main/pkg/metrics/metrics.go#L81-L82).

File: velero-mixin/alerts/velero-alerts.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        expr: delta(velero_restore_failed_total[5m]) > 0
+        for: 1m
+        labels:
+          severity: critical
+        annotations:
+          summary: 'Failed Velero Restore.'
+          description: 'Velero backup {{ $labels.schedule }} has failed Restores.'
+      
+      - alert: NoBackupJob
+        expr: delta(velero_backup_attempt_total[24h]) <= 0
```
[the remark this answers is not in the document]
Comment (ishanjainn) [reply velero-mixin/alerts/velero-alerts.yml:32]: For most of the folks yes, As folks do use velero to backup volumes for their database'es and thats a required daily run. I can make this configurable in the mixin though would still prefer to set it as 1 when creating an integration out of this

**Comment by ishanjainn** [comment]
Thanks Ryan!
With the updated changes I have:
- used increase(metric{...}[$__rate_interval]) and created single panel for backups overview, restore, deletion, snapshots and csi snapshots
- I have still kept the stats boxes uptop just to get lifetime stats, Kinda find it useful
- add the schedule label and variable
- updated the no backup alert to trigger on 48 hours instead of 24hrs
- Introduced an alert for no volume snapshots(includes csi snapshots) in the past 24 hours. Volume snapshots are generally taken daily(sometimes more frequently depending on the position of the database in the pipeline) so hence kept it the same

**Review by rgeyer — APPROVED** [review]
Great work! Thank you for integrating the feedback.

If you rebase/merge with main, it'll fix the CI, it was fixed in #1023 

🚀

**Pull request merged**